kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -708,265 +670,198 @@ class ParallelDistanceCalculator
 
         bool is_distributed = false;
-        if(rModelPart.GetCommunicator().TotalProcesses() > 1)
+        if(rModelPart.GetCommunicator().TotalProcesses() > 1) {
             is_distributed = true;
-
-        const int node_size = rModelPart.Nodes().size();
+        }
 
         //check that variables needed are in the model part
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"distance Variable is not in the model part","");
-
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"Area Variable is not in the model part","");
-
-        if(is_distributed == true)
-            if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX)) )
-                 KRATOS_THROW_ERROR(std::logic_error,"PARTITION_INDEX Variable is not in the model part","")
+        const int node_size = rModelPart.Nodes().size();
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar))) << "Distance variable is not in the model part." << std::endl;
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar))) << "Area variable is not in the model part." << std::endl;
+        if (is_distributed == true) {
+            KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX))) << "Area variable is not in the model part." << std::endl;
+        }
 
 		KRATOS_CATCH("")
 	}
 
-    void ResetVariables(ModelPart& rModelPart,
-                            const Variable<double>& rDistanceVar,
-							const double MaxDistance)
+    void ResetVariables(
+        ModelPart& rModelPart,
+        const Variable<double>& rDistanceVar,
+        const double MaxDistance)
     {
         KRATOS_TRY
 
         //reset the variables needed
-        const int node_size = rModelPart.Nodes().size();
-
-        #pragma omp parallel for
-        for(int i = 0; i<node_size; i++)
-        {
-            ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
+        block_for_each(rModelPart.Nodes(), [&](NodeType& rNode) {
             //it->FastGetSolutionStepValue(rAreaVar) = 0.0;
-            double& dist = it->FastGetSolutionStepValue(rDistanceVar);
-            it->SetValue(rDistanceVar,dist); //here we copy the distance function to the fixed database
-
-            if(dist < 0.0)
-                it->SetValue(IS_FLUID,1.0);
-            else
-                it->SetValue(IS_FLUID,0.0);
+            double& r_dist = rNode.FastGetSolutionStepValue(rDistanceVar);
+            rNode.SetValue(rDistanceVar, r_dist); //here we copy the distance function to the fixed database
 
-            dist = MaxDistance;
+            if(r_dist < 0.0) {
+                rNode.Set(FLUID, true);
+            } else {
+                rNode.Set(FLUID, false);
+            }
```

Comment:
```suggestion
            rNode.Set(FLUID, r_dist < 0.0);
```

I know this was there initially, but code like this makes me cringe :sweat_smile:

Reply:
:smile: I left these for a future refactor/clean up but OK.